.evergreen/download-mongodb.sh

```diff
@@ -267,14 +277,12 @@ get_mongodb_download_url_for ()
              MONGODB_32="http://downloads.mongodb.org/linux/mongodb-linux-x86_64-rhel55${DEBUG}-${VERSION_32}.tgz"
              MONGODB_30="http://downloads.mongodb.org/linux/mongodb-linux-x86_64-rhel55${DEBUG}-${VERSION_30}.tgz"
       ;;
-      linux-sles-11*-x86_64)
```

Comment:
Noted this moved to the bottom of the linux-sles list to be consistent with the ordering of other systems 👍